src/Symfony/Bridge/Doctrine/Listener/DoctrineConnectionListener.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Listener;
+
+use Doctrine\Persistence\ConnectionRegistry;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+final class DoctrineConnectionListener implements EventSubscriberInterface
+{
+    /**
+     * @param \ArrayObject<int, array{name: string, timing: int}> $connectionTiming
+     * @param array<string, int>                                  $timeToLiveByConnection
+     */
+    public function __construct(
+        private readonly \ArrayObject $connectionTiming,
+        private ConnectionRegistry $connectionRegistry,
```

Comment:
This object is too heavy to instantiate.
I was more comfortable with the previous implementation that relied on the string identifier.
Can we find a way that'd use the service ids instead?

Reply:
Why is it heavy ? The ConnectionRegistry used in DoctrineBundle is a wrapper around a ServiceLocator.